packages/relay/src/lib/services/hapiService/hapiService.ts

```diff
@@ -0,0 +1,274 @@
+    this.initialTransactionCount = parseInt(process.env.HAPI_CLIENT_TRANSACTION_RESET!) || 0;
+    this.initialResetDuration = parseInt(process.env.HAPI_CLIENT_DURATION_RESET!) || 0;
+    this.initialErrorCodes = JSON.parse(process.env.HAPI_CLIENT_ERROR_RESET || "[]");
+
+    this.transactionCount = this.initialTransactionCount;
+    this.resetDuration = currentDateNow + this.initialResetDuration;
+    this.errorCodes = this.initialErrorCodes;
+
+    this.isReinitEnabled = true;
+    this.isTimeResetDisabled = this.resetDuration === currentDateNow;
+
+    if (this.transactionCount === 0 && this.errorCodes.length === 0 && this.isTimeResetDisabled) {
+      this.isReinitEnabled = false;
+    }
+    this.shouldReset = false;
+
+    this.register = register;
+    const metricCounterName = 'rpc_relay_client_service';
+    register.removeSingleMetric(metricCounterName);
+    this.clientResetCounter = new Counter({
+      name: metricCounterName,
+      help: 'Relay Client Service',
+      registers: [register],
+      labelNames: ['transactions', 'duration', 'errors'],
+    });
+  }
+
+  /**
+   *  Decrement transaction counter. If 0 is reached, reset the client. Check also if resetDuration has been reached and reset the client, if yes.
+   */
+  private decrementTransactionCounter() {
+    if (this.transactionCount == 0) {
+      return;
+    }
+
+    this.transactionCount--;
+    if (this.transactionCount <= 0) {
+      this.shouldReset = true;
+    }
+  }
+
+  /**
+   *  Decrement error encountered counter. If 0 is reached, reset the client. Check also if resetDuration has been reached and reset the client, if yes.
+   */
+  public decrementErrorCounter(statusCode: number) {
+    if (!this.isReinitEnabled || this.errorCodes.length === 0) {
+      return;
+    }
+
+    if (this.errorCodes.includes(statusCode)) {
+      this.shouldReset = true;
+    }
+  }
+
+  private checkResetDuration() {
+    if (this.isTimeResetDisabled) {
+      return;
+    }
+
+    if (this.resetDuration < Date.now()) {
+      this.shouldReset = true;
+    }
+  }
+
+  /**
+   * Reset the SDK Client and all counters.
+   */
+  private resetClient() {
+    this.clientResetCounter
+      .labels(this.transactionCount.toString(), this.resetDuration.toString(), this.errorCodes.toString())
+      .inc(1);
+
+    this.client = this.initSDKClient(this.logger, this.register);
+    this.resetCounters();
+  }
+
+  /**
+   * Reset all counters with predefined configuration.
+   */
+  private resetCounters() {
+    this.transactionCount = this.initialTransactionCount;
+    this.resetDuration = Date.now() + this.initialResetDuration;
+
+    this.shouldReset = false;
+  }
+
+  /**
+   * Configure SDK Client from main client
+   * @param {Logger} logger
+   * @param {Registry} register
+   * @returns SDK Client
+   */
+  private initSDKClient(logger: Logger, register: Registry): SDKClient {
+    return new SDKClient(this.clientMain, logger.child({ name: `consensus-node` }), register);
```

Comment:
You'll wanna consider the recent but we just fixed https://github.com/hashgraph/hedera-json-rpc-relay/pull/1249